internal/rebooter/reconcile.go

```diff
@@ -27,23 +27,108 @@ var (
 type RebooterReconciler struct {
 	*reconciler.Reconciler
 	reconcileTimeout time.Duration
+	nodeName         string
 }
 
-func NewRebooterReconciler(client client.Client, scheme *runtime.Scheme, recorder record.EventRecorder, reconcileTimeout time.Duration) *RebooterReconciler {
+func NewRebooterReconciler(
+	client client.Client,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+	reconcileTimeout time.Duration,
+	nodeName string,
+) *RebooterReconciler {
 	r := reconciler.NewReconciler(client, scheme, recorder)
 	return &RebooterReconciler{
 		Reconciler:       r,
 		reconcileTimeout: reconcileTimeout,
+		nodeName:         nodeName,
 	}
 }
 
 func (r *RebooterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName(ControllerName)
 	logger.Info(fmt.Sprintf("Reconciling %s", req.Name))
+	node := &corev1.Node{}
+	err := r.Get(ctx, client.ObjectKey{Name: r.nodeName}, node)
+	if err != nil {
+		return ctrl.Result{RequeueAfter: r.reconcileTimeout}, fmt.Errorf("failed to get node %s: %w", r.nodeName, err)
+	}
+
+	if r.checkIfNodeNeedsReboot(ctx, node) {
+		logger.Info("Node needs reboot")
+
+		if err := r.RebootNode(ctx, node); err != nil {
+			return ctrl.Result{RequeueAfter: r.reconcileTimeout}, fmt.Errorf("failed to reboot node %s: %w", r.nodeName, err)
+		}
+	}
+
+	if r.checkIfNodeNeedsDrain(ctx, node) {
+		logger.Info("Node needs drain")
+
+		if err := r.DrainNode(ctx, node); err != nil {
+			return ctrl.Result{RequeueAfter: r.reconcileTimeout}, fmt.Errorf("failed to drain node %s: %w", r.nodeName, err)
+		}
+	}
```

Comment:
now it looks like if node has 2 conditions `Drain` and `Reboot`, it'll first be rebooted, and then drained, but the remaining drain looks like not needed

maybe it'll be done later in implementation of `RebootNode` that node Conditions (`Drain` and `Reboot`) will be cleaned somehow?